subprojects/core-api/src/main/java/org/gradle/api/artifacts/VersionCatalog.java

```diff
@@ -50,4 +51,28 @@
      * @param name the name of the version
      */
     Optional<VersionConstraint> findVersion(String name);
+
+    /**
+     * Returns the list of aliases defined in this version catalog.
+     * @return the list of dependency aliases
+     *
+     * @since 7.1
+     */
+    List<String> getDependencyAliases();
```

Comment:
Why not expose the `Map` directly?
I understand it might require more work internally because of the mapping required. But this could be done lazily on map value access only.
And it would give users a better API that can be used with a single iteration instead of having to iterate and look up always.

Reply:
🤔 I see what you are saying. But the moment someone calls this method, they are looking for bulk handling anyway it seems.
I am fine to go with this for now.